planetiler-custommap/src/main/java/com/onthegomap/planetiler/custommap/ConfiguredFeature.java

```diff
@@ -0,0 +1,254 @@
+package com.onthegomap.planetiler.custommap;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.BiConsumer;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import com.onthegomap.planetiler.FeatureCollector;
+import com.onthegomap.planetiler.FeatureCollector.Feature;
+import com.onthegomap.planetiler.custommap.configschema.AttributeDefinition;
+import com.onthegomap.planetiler.custommap.configschema.FeatureGeometryType;
+import com.onthegomap.planetiler.custommap.configschema.FeatureItem;
+import com.onthegomap.planetiler.custommap.configschema.TagCriteria;
+import com.onthegomap.planetiler.custommap.configschema.ZoomConfig;
+import com.onthegomap.planetiler.custommap.configschema.ZoomFilter;
+import com.onthegomap.planetiler.geo.GeometryException;
+import com.onthegomap.planetiler.reader.SourceFeature;
+import com.onthegomap.planetiler.util.ZoomFunction;
+
+/**
+ * A map feature, configured from a YML configuration file
+ */
+public class ConfiguredFeature {
+
+  private Collection<String> sources;
+  private Predicate<SourceFeature> geometryTest;
+  private Function<FeatureCollector, Feature> geometryFactory;
+  private Predicate<SourceFeature> tagTest;
+  private BiConsumer<SourceFeature, Feature> zoomConfig;
```

Comment:
These could be final since they are only set once from the constructor. I don't want to clutter the review with comments like this so I'm also looking into linters in #124 that add annotations to the PR like an IDE would show a warning - if you have any experience with or preference on those, let me know!

Reply:
I would personally recommend sonarlint!

Reply:
Fixed specific issue noted here in 97ba5b4